### ISSUE TITLE
Object prototype undefinded

### ISSUE THREAD

**Issue by 1jerry** [alexeyraspopov/dataclass#10]
No matter what I do, including the exact few lines of TypeScript example for User, I get the following error when the program runs, either through ts-node or compile JS code:
```
> [eval]:7
        return extendStatics(d, b);
               ^
TypeError: Object prototype may only be an Object or null: undefined
    at setPrototypeOf (<anonymous>)
```
TS 3.5.2
node 8.9.3 & 11.12.0

**Comment by alexeyraspopov** [comment]
@1jerry, thanks for reporting. Can I ask you to provide more code in order to reproduce the issue? Thanks.

**Comment by alexeyraspopov** [comment]
@1jerry, can I ask you about your environment? What do you do in order to see such issue?

**Comment by 1jerry** [comment]
I type this in:
```
import Record from 'dataclass'

class User extends Record<User> {
    name: string = 'anon';
    age: number = 0;
```
I then type tsc on this script.  That's it.
TS 3.5.2, node 11.12.0 (plus others), Macbook Pro
(I do have a lot of other libraries loaded in package.json, but obviously don't include any in this script.  I can try on a clean environment next week)

**Comment by 1jerry** [comment]
## More details

1. blank directory
1. npm init -y
1. npm i typescript
1. npm i dataclass
1. create file test.ts:
```
import Record from 'dataclass'

class User extends Record<User> {
    name: string = 'anon';
    age: number = 0;
}
```
6. tsc test.ts
1. node test  <--- this is where the error shows up, exactly as shown on my first comment.

**Comment by alexeyraspopov** [comment]
I was able to reproduce the issue, thanks!

What I found is that TS has hard time trying to import `Record` from node modules. As a Common JS module, `dataclass` exports `Record` as a default export. If you check compiled TS code, you can find that it attempts to use this default export as 

```
var dataclass_1 = require("dataclass");
var Record = dataclass_1["default"];
```

I'm trying to find explanation/solution/workaround for this behavior.

**Comment by alexeyraspopov** [comment]
@1jerry, can there be something wrong with typings https://github.com/alexeyraspopov/dataclass/blob/master/typings/dataclass.d.ts#L4? I'm not a TS user.

**Comment by 1jerry** [comment]
I tried several things but I didn't discover a clue as to why this error is
happening.
-- 
Jerry
--

**Comment by alexeyraspopov** [comment]
I don't know if the original issue has been resolved or there were any idea where it is coming from. I've just published the next major release `dataclass@2.0.0` which now uses named imports instead of the default import. This should definitely help with the described issue, basically avoiding it.

```ts
// Note that Record is now called Data
// Named export should help with possible CJS/ESM compatibility issues in any environment
import { Data } from "dataclass";

class Entity extends Data {
  counter: number = 0;
}
```

I hope you'll have a chance to check it out. Please let me know if you face any issues.
